motionplan/armplanning/real_test.go

```diff
@@ -142,3 +144,93 @@ func TestSandingLargeMove1(t *testing.T) {
 
 	test.That(t, len(solution.steps), test.ShouldEqual, 1)
 }
+
+func TestPirouette(t *testing.T) {
+	t.Skip()
+
+	// get arm kinematics for forward kinematics
+	armName := "ur5e"
+	armKinematics, err := referenceframe.ParseModelJSONFile(utils.ResolveFile("components/arm/fake/kinematics/ur5e.json"), armName)
+	test.That(t, err, test.ShouldBeNil)
+
+	idealJointValues := [][]referenceframe.Input{
+		{{0 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{30 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{60 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{90 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{120 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{150 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{180 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{180 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{150 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{120 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{90 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{60 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{30 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{0 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+	}
+	// the only change here is in joint 0 in increments of 30, while all the other joints are kept at a constant value
+	// below is change in joint 0 in degrees:
+	// 0 -> 30 -> 60 -> 90 -> 120 -> 150 -> 180 -> 180 -> 150 -> 120 -> 90 -> 60 -> 30 -> 0
+
+	// determine pose given elements of idealJointValues
+	pifs := []*referenceframe.PoseInFrame{}
+	for _, pos := range idealJointValues {
+		pose, err := armKinematics.Transform(pos)
+		test.That(t, err, test.ShouldBeNil)
+		posInF := referenceframe.NewPoseInFrame(referenceframe.World, pose)
+		pifs = append(pifs, posInF)
+	}
+
+	// construct framesystem
+	fs := referenceframe.NewEmptyFrameSystem("pirouette")
+	err = fs.AddFrame(armKinematics, fs.World())
+	test.That(t, err, test.ShouldBeNil)
+
+	for iter := 0; iter < 10; iter++ {
```

Comment:
This should stay in to combat flakiness when this is un-skipped.
Running without this loop results in test passing sometimes and sometimes failing.